packages/broker/src/plugins/operator/ContractFacade.ts

```diff
@@ -0,0 +1,371 @@
+import { Provider } from '@ethersproject/providers'
+import { Operator, Sponsorship, operatorABI, sponsorshipABI } from '@streamr/network-contracts'
+import { StreamID, ensureValidStreamPartitionIndex, toStreamID } from '@streamr/protocol'
+import { EthereumAddress, Logger, TheGraphClient, addManagedEventListener, toEthereumAddress } from '@streamr/utils'
+import { Contract } from 'ethers'
+import sample from 'lodash/sample'
+import fetch from 'node-fetch'
+import { NetworkPeerDescriptor } from 'streamr-client'
+import { OperatorServiceConfig } from './OperatorPlugin'
+
+interface RawResult {
+    operator: null | { latestHeartbeatTimestamp: string | null }
+}
+
+interface EarningsData {
+    sponsorshipAddresses: EthereumAddress[]
+    sumDataWei: bigint
+    maxAllowedEarningsDataWei: bigint
+}
+
+export const VOTE_KICK = '0x0000000000000000000000000000000000000000000000000000000000000001'
+export const VOTE_NO_KICK = '0x0000000000000000000000000000000000000000000000000000000000000000'
+
+export class ParseError extends Error {
+    public readonly reasonText: string
+
+    constructor(reasonText: string) {
+        super(`Failed to parse metadata: ${reasonText}`)
+        this.reasonText = reasonText
+    }
+}
+
+export function parsePartitionFromReviewRequestMetadata(metadataAsString: string | undefined): number | never {
+    if (metadataAsString === undefined) {
+        throw new ParseError('no metadata')
+    }
+
+    let metadata: Record<string, unknown>
+    try {
+        metadata = JSON.parse(metadataAsString)
+    } catch {
+        throw new ParseError('malformed metadata')
+    }
+
+    const partition = Number(metadata.partition)
+    if (isNaN(partition)) {
+        throw new ParseError('invalid or missing "partition" field')
+    }
+
+    try {
+        ensureValidStreamPartitionIndex(partition)
+    } catch {
+        throw new ParseError('invalid partition numbering')
+    }
+
+    return partition
+}
+
+export type ReviewRequestListener = (
+    sponsorship: EthereumAddress,
+    operatorContractAddress: EthereumAddress,
+    partition: number
+) => void
+
+const logger = new Logger(module)
+
+export interface SponsorshipResult {
+    sponsorshipAddress: EthereumAddress
+    streamId: StreamID
+    operatorCount: number
+}
+
+export class ContractFacade {
```

Comment:
Yes. We chose not to use that pattern so that each class has all of its fields listed in one place. Then it is very easy to check what fields any class has. Usually the complete list of fields in a class tells very much about a class, and is an efficient way to get familiar what a class does.

E.g. here someone could easily misread that the class has only one field:
```
class Foo {

     private foo: number = 0
     
     constructor(private bar: number) {}

}
```

Reply:
We have an eslint rule where the preference is to declare the fields explicit.

Reply:
Cool.